app/modules/FrontModule/Registration/components/RegistrationForm/RegistrationForm.php

```diff
@@ -45,7 +44,8 @@ public function createComponentRegistrationForm(): Form
     {
         $form = new Form();
 
-        $form->addHidden('email');
+        $form->addHidden('email', '');
+        $form->addHidden('year', '2020');
```

Comment:
tohle bych jako input z webu nedaval. Snadno to zmenis a nikde se to nekontroluje ze to ma byt 2020

Reply:
Toto je nejjednodušší cesta. Jasně, může to být změněno, ale stejně tak můžeš například natvrdo nastavit value, nebo přidat ověření. Nicméně, když už chceš používat 'setMappedType', tak je lepší, když ti vše rovnou naplní formulář, včetně nějakých konstant, než to pak někde ověřovat, jestli se to naplnilo a pokud ne tak tam dát 2020 ručně. Pokud jde o kód, tak jsem se ho snažil co nejméně měnit. Ale například zrovna když už, tak ta 2020 by měla být někde konstanta a v db by mělo být integritní omezení aby nešlo vložit nic jiného, nebo ještě lépe by se to nemělo v aplikaci vůbec posílat nikam, tedy žádné nastavování v entitě (možná max default value - * @property int $year {default 2020}), žádné hidden inputy ve formuláři, ale pěkně výchozí hodnota na straně DB, tím ti odpadne hromada nepříjemností.

Udělal jsem to takto, jelikož jsi to plnil v kódu a chtěl jsem to zachovat co nejblíže tomu, jak to bylo napsáno.

Reply:
Jak psal Honza, tohle není moc fajn. Chápu tvůj záměr, ale hidden input mi nepřijde jako fajn cesta. A defaultní hodnota v DB též ne, když NetteCamp je každý rok.

Vzhledem k tomu že ta třída do které to dáváš je v podstatě DTO (Data Transfer Object), proč si prostě nepředat hodnotu z konfigu ve které by byl rok a jen to dodat ručně (`$dto->year = 2020;`) do DTO?

Reply:
Jelikoz o ukladani se stara nejaka servisa RegistrationService ? (snad :) ) tak by to mohla ona dostat parametrem z configu. Ale na 99.9% to je stejne date("Y"). Jelikoz registraci spoustime nekdy v unoru daneho roku. Drive to nema cenu...

Reply:
**Vojta:** Tak teoreticky by mohla být nějaká konfigurační tabulka, která by měla uchováno číslo roku konání následujícího Nette campu a toto číslo by se vždy incrementovalo samo po skončení možnosti registrace s tím, že by se od té doby mohla automaticky spustit registrace na další ročník? 

**Chemix:** Tím pádem pokud chcete spouštět registraci až v novém roce, pak by mohlo stačit opravdu jak jsem psal výchozí hodnota v databázi: 'YEAR(CURDATE())', nebo '{default date('Y')}' v definici entity.